packages/frontend/src/components/wallet/NFTBox.js

```diff
@@ -130,12 +130,22 @@ const NFTBox = ({ tokenDetails }) => {
             {
                 ownedTokensMetadata &&
                 <div className='tokens'>
-                    {ownedTokensMetadata.map(({ token_id, metadata: { mediaUrl, title } }) => {
+                    {ownedTokensMetadata.map(({ token_id, metadata: { mediaUrl, title } }, index) => {
+                        const videoProps = index === 0 ? { autoplay: "true" } : {};
                         return <div className='nft' key={token_id}>
-                            <img src={mediaUrl} alt='NFT' onError={(e) => {
-                                e.target.onerror = null;
-                                e.target.src = FailedToLoad;
-                            }}/>
+                            {
+                                mediaUrl.match(/\.webm$/i)
+                                    ? <video muted="true" loop controls { ...videoProps }>
+                                        <source src={mediaUrl} type="video/webm" onError={(e) => {
+                                        e.target.onerror = null;
+                                        e.target.src = FailedToLoad;
```

Comment:
I noticed that `FailedToLoad` is an SVG image, but this is a `<video>` tag -- have you tested that this works and correctly displays our `FailedToLoad` SVG if the .webm file cannot be loaded by the browser?

Reply:
So yes, you are correct in that a "Loading..." svg would be much better.

Initially I did just copy the logic from the img tag using the video error handler, however after receiving your feedback and testing that condition, I discovered that the video tag will not render images...

I do feel that if the nft creator's intent is to show a video, the error condition should relate to that intent?  What do you think?  So, in that sense, a video fallback ("Loading..." is probably fine.

Reply:
So I think we have two different issues here -- one is what to display while the video is being loaded, and the other is what to display if the video fails to load.

**I think that the behaviour here should be:**
1. When the page loads, the user sees 'Loading' image for the video until it is ready to be played.
2. If the video fails to load, the user sees the 'FailToLoad' image.
3. If the video loads successfully ('canplay' event fires), then the 'loading' image changes to show the default preview of the video (probably the first frame of the video in most browsers)

I don't think setting `poster` to the loading _or_ the 'loading failed' image will work for us, because it will show. the poster image before the user actually taps 'play' on the the video to download it, and they will likely think they must wait until the `loading` image changes to show something else before tapping 'play', or if we show the error image, they will assume it has failed to load even though it is still pending.

I did some reading about handling this, and I think the pattern described here is what we want, but adapted to be a react-based solution so we don't touch the DOM directly of course: https://stackoverflow.com/a/24470253

**Implementation-wise, I think we can get the behaviour pattern I described above by:**
1. Set the `poster` property to a transparent gif (it can be in-line)
2. Set an initial class on the video tag that sets the background to 'Loading...'
3. Attach a `canplay` event handler to the appropriate (video or source nested in the video) tag so that it removes the `loading` class _and_ sets the `poster` to empty; this way the user will see a proper preview of the video as soon as it is in a state that it can be played.
4. Attach an `error` event handler to the `<source>` tag inside the `<video>` tag (see the post that I referenced in my first comment, above in this thread).  If that handler fires, we set a class on the `video` tag that sets the background of the video tag to our `FailedToLoad` SVG _and_ re-configures the `poster` to be the blank gif.

What do you think?